docs/source/examples/BSTLD.rst

```diff
@@ -1,288 +1,182 @@
-The :ref:`code block <BSTLD-dataloader>` below displays the ``BSTLD`` dataloader.
+The second step is to write the :ref:`reference/glossary:Dataloader`.
+The function of :ref:`reference/glossary:Dataloader` is to read the dataset into a
+:ref:`reference/dataset_structure:Dataset` object.
+The :ref:`code block <BSTLD-dataloader>` below displays the "BSTLD" dataloader.
 
 .. literalinclude:: ../../../tensorbay/opendataset/BSTLD/loader.py
    :language: python
    :name: BSTLD-dataloader
    :linenos:
-   :emphasize-lines: 11-12,52
+   :emphasize-lines: 11-12,58
 
-There are mainly two steps to write a :ref:`contribution:Dataloader`:
-
--  Create a :ref:`basic_concepts:Dataset` and its relevant :ref:`Segments <basic_concepts:Segment>`.
--  Add the :ref:`basic_concepts:Data` and corresponding Labels
-   to the created :ref:`Segments <basic_concepts:Segment>`.
-
-Create Dataset and Segments
----------------------------
-
-Note that after creating the :ref:`basic_concepts:Dataset`,
-you need to load the :ref:`contribution:Catalog`.(L52)
+Note that after creating the :ref:`reference/dataset_structure:Dataset`,
+you need to load the :ref:`reference/dataset_structure:catalog`.(L58)
 The catalog file ``catalog.json`` is in the same directory with dataloader file.
 
 In this example, we create segments by ``dataset.create_segment(SEGMENT_NAME)``.
 You can also create a default segment without giving a specific name, then its name
 will be "".
 
-Add Data and Labels
--------------------
-
-It takes four key substeps to add the data and the labels:
-
--  Creating :ref:`basic_concepts:Data` and adding content.
--  Creating :ref:`supported_label_types:Box2D` and adding annotations.
--  Appending the Box2D labels to the created Data.
--  Appending the Data to the created Segment.
-
-See :ref:`this page <supported_label_types:Box2D>` for more details for about Box2D annotation details.
+See :ref:`this page <reference/label_format:Box2D>` for more details for about Box2D annotation details.
 
 .. note::
    The :ref:`BSTLD dataloader <BSTLD-dataloader>` above uses relative import(L11-12).
-   However, when you write your own dataloader you should use regular import as shown below.
+   However, when you write your own dataloader you should use regular import.
    And when you want to contribute your own dataloader, remember to use relative import.
 
-.. code:: python
+*****************************
+ Upload Images to the Dataset
+*****************************
+
+After you finish the :ref:`reference/glossary:Dataloader` and organize the "BSTLD" into a Dataset class,
+:ref:`reference/dataset_structure:Dataset` object, you can upload it to TensorBay to share with the community.
+
+.. literalinclude:: ../../../examples/bstld.py
+   :language: python
+   :lines: 27-31
+
+*****************************
+ Read Images from the Dataset
+*****************************
+
+After uploading the "BSTLD" dataset, you can now read it from TensorBay.
```

Comment:
After uploading the "BSTLD" dataset, you can now read it from TensorBay. -> Now you can read "BSTLD" dataset from TensorBay.